usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
I think the "some use cases modify pool state a lot so this will be expensive" kinda misses the fundamental question. Should ZFS support an efficient way for a user space application to be able to monitor these events? If no, please explain why this is an unreasonable requirement. If yes,  and you don't like this strategy, please offer another one. :-)